File: source-connector/src/main/java/sourceconnector/domain/log/FileLogMetadata.java
package sourceconnector.domain.log;

import offsetmanager.domain.file.FileKey;

/**
 *
 * @param fileKey the file path log is saved
 * @param offset offset in the log file
 */
public record FileLogMetadata(
  FileKey fileKey,
  long offset
) implements LogMetadata {
  @Override
  public String key() {
    return fileKey.get();
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/log/FileLogMetadata.kt
package sourceconnector.domain.log

import offsetmanager.domain.file.FileKey

/**
 *
 * @param key the file path log is saved
 * @param offset offset in the log file
 */
data class FileLogMetadata(
  override val key: FileKey,
  override val offset: Long
) : LogMetadata {
}
